### ISSUE TITLE
DELETED REPOS THAT WERE NOT SELECTED

### ISSUE THREAD

**Issue by LSleutsky** [moollaza/repo-remover#173]
I realize this is an old repo, and I should have been more attentive, but this "app" deleted most of my repos that were not selected in the list. I specifically made sure they were not selected, yet they are now all gone.

Great app, thanks for the fuckery!

**Comment by moollaza** [comment]
@LSleutsky oh no! I'm very, very sorry to hear that. Definitely not the intention of the app.

The repo is a few years old, but the code should still be working fine. Others are using it daily.

I'm not sure how this would be possible, beyond a bug in the code. Tons of people have used RepoRemover, but this is the first I've heard of something like this happening.

There should have been a confirmation screen listing all the repos that were going to be deleted -- can you confirm if they were NOT listed there?

Did you happen to select and unselect the repos that were accidentally deleted? I'm trying to understand how this could have happened (e.g. a bug in state handling -- but again it should have shown whatever was going to be deleted).

I've been slowly re-writing the app in Svelte as I'm out of touch with Vue.js and will add some tests to make sure this isn't possible.

**Comment by LSleutsky** [comment]
Can 150% without a shadow of a doubt confirm that the repos were not listed in the confirmation screen's list of to-be-deleted repos. No doubt!


Luckily GitHub support was able to restore all of my repos that were deleted, thank god.

**Comment by moollaza** [comment]
> Luckily GitHub support was able to restore all of my repos that were deleted, thank god.

Glad to hear that. Apologies for any trouble.

---

I'll play around a bit to see if I can re-create what happened. 

If you don't mind, and can remember, I'd appreciate if you could list the steps taken (e.g. changing filters, entering a search query, selecting/unselecting repos)

**Comment by LSleutsky** [comment]
All I did after creating a valid token was see my repositories in a list. I had only 5 per page, and would individually select the checkbox for what I wanted deleted. Then a confirmation would screen modal would pop up, listing the repositories I just checked, asking me to enter my GitHub user name to confirm deletion. Which I did.


I went thru this process a considerable amount of times because I had many repos to go thru, and specifically filtered only 5 per page in the repo remover app, to be more attentive as to what was being deleted...
